src/epoch3d.F90

```diff
@@ -136,10 +139,20 @@ PROGRAM pic
 #endif
 
     step = step + 1
-    time = time + dt
```

Comment:
Is this repositioning of the time increment deliberate?

For core EPOCH (with time-dependent EM boundary conditions), the time increase `time = time + dt` must happen before `CALL update_eb_fields_final` to enable the correct boundary condition to be applied.

Reply:
Sorry, wasn't being very careful.

I've fiddled with it now, but I think each of the different sub-stepping methods might benefit from maintaining their own internal concept of where in the timestep they are, rather than having it as a global.

Reply:
Yeah. I couldn't work out what the correct solution was for the different solver options either. Agree it's worth considering each solver maintaining their own step.